Chapter on maintaining consistency in The Turing Way chapters (#1459)
alan-turing-institute/the-turing-way: book/website/community-handbook/consistency.md

(ch-consistency)=

> # Achieving Retrospective Consistency

Reviewer: Hi @paulowoicho!! As we discussed in person, could we change this landing page for the chapter to be a little more "timeless". Something like "Maintaining consistency" perhaps? The somewhere in the chapter you can acknowledge the work you're doing (and how you're doing it - writing guidelines, writing issues etc) rather than making the _retrospective_ aspect the headline of the chapter?

These checks greatly enhance the overall look and feel of the book but can be safely ignored.

An overview of these requirements is itemised in {ref}`below <ch-consistency-requirements-checklist>`. The rest of this chapter explains them in more detail and describes how they can be achieved.

> Please note that these requirements are not exhaustive or definitive, and neither are the classifications rigid. 
> If you identify more consistency issues that need to be addressed, join the conversation [here](https://github.com/alan-turing-institute/the-turing-way/issues/1174). 
> There is also no order to the items in the checklist. 
> No item is inherently more important than the other.

(ch-consistency-requirements-checklist)=
### Consistency Checklist

> This checklist was developed based on the insights gained from _The Turing Way's_ participation in 2020's [Google Season of Docs](https://developers.google.com/season-of-docs) (GSoD) Programme.
> The [Technical Writer](https://github.com/paulowoicho) selected to work on _The Turing Way_ helped make the book retrospectively consistent by creating guidelines, templates, and issues to facilitate new contributions.

> > The PRs and issues raised by the Technical Writer are tagged with the `gsod2020` label in _The Turing Way_ Github repo.

Author: Ahh okay. I have reworded it to only refer to The Turing Way's participation in GSoD. I had included this to address @KirstieJane  comments earlier.

> > The PRs and issues raised by the Technical Writer are tagged with the `gsod2020` label in _The Turing Way_ Github repo.

Reviewer 2: Generally, a chapter is open to change and development. This consistency chapter will also be changed as our procedures evolve over time. With that in mind, we do not add authorship on any chapter, but allow people to describe their notable contributions in the `contributors.md` without taking away the work they do. Furthermore, we will be creating a report for GSoD where all your contributions should be described including this chapter. In your case as well I would suggest to stick to the standard process unless I missed something you discussed with @KirstieJane - in which case, do let me know.

- Hard Requirements
  - Add labels to chapters, subchapters and images to enable cross-referencing.
  - Convert all HTML formatting to Markdown.
  - Convert image formatting from Markdown to `MyST`.

>   - Remove empty files from the `_toc.yml`.

Reviewer 2: Move the _toc file point to the end of the hard requirement list

  - Ensure chapters follow a consistent structure.
  - Ensure chapters are consistent with the Turing Way Style Guide.
  - Ensure correct grammar and a consistent tone across the book.

>   - Ensure chapters use a consistent language.

Reviewer 2: ```suggestion
  - Ensure chapters use a consistent language, for example, if you choose to write in British English, maintain that throughout
```

>   - Add ALT text to images.

Reviewer 2: ```suggestion
```

Reviewer 2: Moved this to a previous point about the image

>   - Fix Markdown formatting of non-consecutive headers

Reviewer 2: This can move to the top

>   - Remove ToCs from chapter body since one is auto-generated by `jupyter-books`.

Reviewer 2: ```suggestion
  - Do not add a 'table of content' in chapters or subchapters as it is auto-generated by the Jupyter Book
```

- Soft Requirements

>   - Ensure each subchapter has a good summary in their landing page.

Reviewer 2: ```suggestion
  - Ensure each chapter has a good summary on its landing page along with links to its subchapters.
```

  - Split long chapters into smaller subchapters so they are modular.

>   - Ensure the names of chapters/subchapters map exactly to the left ToC.

Reviewer 2: ```suggestion
  - Ensure that the names of chapters/subchapters are short and map exactly to how they are titled in the `_toc.yml`.
```

>   - Ensure proper title-casing for headers.

Reviewer 2: Add an example here, is it supposed to be the camel case or only the first letter capital

These checks greatly enhance the overall look and feel of the book but can be safely ignored if you can't manage to integrate them into your contribution.
You can always return to review your past contributions and enhance their content.

An overview of these requirements is itemised {ref}`below <ch-consistency-requirements-checklist>`. The rest of this chapter explains them in more detail and describes how they can be achieved.

> Please note that these requirements are not exhaustive or definitive, and neither are their classifications rigid. 
> If you identify more consistency issues that need to be addressed, join the conversation [here](https://github.com/alan-turing-institute/the-turing-way/issues/1174). 
> There is also no order to the items in the checklist. 
> No item is inherently more important than the other.

(ch-consistency-requirements-checklist)=
### Consistency Checklist

> This checklist was developed based on the insights gained from _The Turing Way's_ participation in 2020's [Google Season of Docs](https://developers.google.com/season-of-docs) (GSoD) Programme.

> > The [Technical Writer](https://github.com/paulowoicho) selected to work on _The Turing Way_ helped make the book retrospectively consistent by creating guidelines, templates, and issues to facilitate new contributions.

Author: ```suggestion
> The Technical Writer selected to work on _The Turing Way_ helped make the book retrospectively consistent by creating guidelines, templates, and issues to facilitate new contributions.
```

(ch-consistency)=
# Maintaining Consistency

> _The Turing Way_ project is an open-source endeavour that empowers contributors from around the world to leverage their skills, knowledge and expertise to build and expand its content.

Reviewer 3: ```suggestion
_The Turing Way_ project is an open-source endeavour that empowers contributors from around the world to leverage their skills, knowledge, and expertise to build and expand its content.
```

> _The Turing Way_ project is an open-source endeavour that empowers contributors from around the world to leverage their skills, knowledge and expertise to build and expand its content.

Reviewer 3: ```suggestion
_The Turing Way_ is an open-source project that empowers contributors around the world to leverage their skills, knowledge, and expertise to build and expand its content.
```

> _The Turing Way_ guides are evolving with multiple chapters co-developed by individuals from varied backgrounds, who are all passionate about sharing knowledge around data science and research.

Reviewer 3: ```suggestion
_The Turing Way_ guides are continually evolving, with multiple chapters co-developed by individuals from varied backgrounds - who are all passionate about sharing knowledge around data science and research.
```

> To sustain and support this vibrant community, _The Turing Way_ book must remain consistent and accessible as it evolves. 

Reviewer 3: ```suggestion
To sustain and support this vibrant community, _The Turing Way_ book should remain consistent and accessible as it evolves. 
```

> Although the {ref}`ch-style-guide` chapter already outlines guidelines for keeping a consistent style across the book, not all chapters follow these suggestions as they have been written asynchronously by different authors. 

Reviewer 3: ```suggestion
The {ref}`ch-style-guide` chapter already provides guidelines for maintaining a consistent style across the book. However, not all chapters follow these suggestions since they're often written asynchronously by different authors. 
```

(ch-consistency)=
# Maintaining Consistency

_The Turing Way_ is an open-source project that empowers contributors around the world to leverage their skills, knowledge, and expertise to build and expand its content.
_The Turing Way_ guides are continually evolving, with multiple chapters co-developed by individuals from varied backgrounds - who are all passionate about sharing knowledge around data science and research.
To sustain and support this vibrant community, _The Turing Way_ book should remain consistent and accessible as it evolves. 
The {ref}`ch-style-guide` chapter already provides guidelines for maintaining a consistent style across the book. However, not all chapters follow these suggestions since they're often written asynchronously by different authors. 
This calls for a need to encourage and support our contributors to maintain consistency throughout the chapters in _The Turing Way_ guides. 

> Achieving this goal alone is reasonably challenging. 

Author: Yeah I think that's what Malvika was saying here. To highlight that one person cant make the Turing Way book consistent and that it takes the effort of several people

> Achieving this goal alone is reasonably challenging. 

Reviewer 3: Got it, I'll make a quick edit back in the file :)

Hard requirements are essential consistency checks that a chapter must pass so that _The Turing Way_ builds without errors, and is readable and accessible by everyone.

Soft requirements, on the other hand, are nice-to-have consistency checks that a chapter should pass. 
These checks greatly enhance the overall look and feel of the book but can be safely ignored if they cannot be integrated into your contribution.
You can always return to review your past contributions and enhance their content.

An overview of these requirements is itemised {ref}`below <ch-consistency-requirements-checklist>`. The rest of this chapter explains them in more detail and describes how they can be achieved.

```{important} Please note that these requirements are not exhaustive or definitive, and neither are their classifications rigid.

If you identify more consistency issues that need to be addressed, join the conversation [here](https://github.com/alan-turing-institute/the-turing-way/issues/1174).

No item is inherently more important than the other.
```

(ch-consistency-requirements-checklist)=
### Consistency Checklist

> #### Hard Requirements

Reviewer 3: ```suggestion
#### Formatting
```

> ##### Formatting

Reviewer 3: ```suggestion
TYPE | REQUIREMENT | CHECK  
---- | ----------- | ----- 
Formatting | Hard | Use Markdown for creating your content (see this [WordPress cheatsheet](https://wordpress.com/support/markdown-quick-reference/)).
Formatting | Hard | Use the headers in sequential order. For example, starting the top level with h1 tag `#`, second-level header with h2 tag `##` and so on.
Formatting | Hard | Add labels to chapters, subchapters and images to enable cross-referencing as described in the {ref}`style guide<ch-style-crossref>` 
Formatting | Hard | Use `MyST` for image formatting as described in the {ref}`style guide<ch-style-figures>` 
Formatting | Soft | Ensure that the names of chapters/subchapters are short and map exactly to how they are titled in the `_toc.yml` 
Formatting | Soft | Ensure proper title-casing for headers | Capitalise the first, last and 'important' words of every title; for example, 'Snow White and the Seven Dwarves'. 
```

> - Use Markdown for creating your content (see this [cheatsheet from WordPress](https://wordpress.com/support/markdown-quick-reference/))

Reviewer 3: ```suggestion
```

>   - Avoid any HTML formatting, or remove all HTML tags when reviewing existing chapters

Reviewer 3: ```suggestion
```

> - Use the headers in sequential order (for example, starting the top level with h1 tag `#`, second-level header with h2 tag `##` and so on)

Reviewer 3: ```suggestion
```

> - Add labels to chapters, subchapters and images to enable cross-referencing as described in the {ref}`style guide<ch-style-crossref>`

Reviewer 3: ```suggestion
```

> - Use `MyST` for image formatting as described in the {ref}`style guide<ch-style-figures>`

Reviewer 3: ```suggestion
```

>   - Use public domain images that are less than 1MB in size and cite appropriately

Reviewer 3: ```suggestion
```

>   - Add a descriptive ALT text to images to allow screen reading software to describe images well to its users

Reviewer 3: ```suggestion
```

##### Structure

>   - Ensure chapters follow a structure as described in this [chapter template](https://github.com/alan-turing-institute/the-turing-way/blob/master/templates/CHAPTER_TEMPLATE.md)

Reviewer 3: ```suggestion
TYPE | REQUIREMENT | CHECK 
---- | ----------- | ----- 
Structure | Hard | Ensure chapters follow a structure as described in this [chapter template](https://github.com/alan-turing-institute/the-turing-way/blob/master/templates/CHAPTER_TEMPLATE.md)
Structure | Hard | Do not add a 'table of contents' in chapters or subchapters as it is auto-generated by the Jupyter Book
Structure | Hard | Ensure external sources are properly referenced and included in the centralised bibtex file as recommended in the {ref}`style guide<ch-style-citing>`
Structure | Hard | Do not add any empty files in the `_toc.yml`, instead create an issue for new chapters
Structure | Soft | Ensure each chapter has a good summary on its landing page along with links to its subchapters.
Structure | Soft | Split long chapters into smaller subchapters so they are modular.
```

>   - Do not add a 'table of content' in chapters or subchapters as it is auto-generated by the Jupyter Book

Reviewer 3: ```suggestion
```

>   - Ensure external sources are properly referenced and included in the centralised bibtex file as recommended in the {ref}`style guide<ch-style-citing>`

Reviewer 3: ```suggestion
```

>   - Do not add any empty files in the `_toc.yml`, instead create an issue for new chapters

Reviewer 3: ```suggestion
```

##### Language

> - Ensure chapters do not use any Latin abbreviation as discussed in the {ref}`style guide<ch-style-guide>`

Reviewer 3: ```suggestion
TYPE | REQUIREMENT | CHECK 
---- | ----------- | ----- 
Language | Hard | Ensure chapters do not use any Latin abbreviation as discussed in the {ref}`style guide<ch-style-guide>`
Language | Hard | Ensure correct grammar and a consistent tone across the book with the help of 1-2 reviewers
Language | Hard | Ensure chapters use a consistent language, for example, if you choose to write in British English, maintain that throughout
```

> - Ensure correct grammar and a consistent tone across the book with the help of 1-2 reviewers

Reviewer 3: ```suggestion
```

> - Ensure chapters use a consistent language, for example, if you choose to write in British English, maintain that throughout

Reviewer 3: ```suggestion
```

> #### Soft Requirements

Reviewer 3: ```suggestion
```

> ##### Formatting

Reviewer 3: ```suggestion
```

> ##### Formatting

Reviewer 3: ```suggestion
```

> - Ensure that the names of chapters/subchapters are short and map exactly to how they are titled in the `_toc.yml`

Reviewer 3: ```suggestion
```

> - Ensure proper title-casing for headers

Reviewer 3: ```suggestion
```

>   - Capitalise the first, last and 'important' words of every title; for example, 'Snow White and the Seven Dwarves'.

Reviewer 3: ```suggestion
```

> ##### Structure

Reviewer 3: ```suggestion
```

>   - Ensure each chapter has a good summary on its landing page along with links to its subchapters.

Reviewer 3: ```suggestion
```

>   - Split long chapters into smaller subchapters so they are modular

Reviewer 3: ```suggestion
```

(ch-consistency)=
# Maintaining Consistency

_The Turing Way_ is an open-source project that empowers contributors around the world to leverage their skills, knowledge, and expertise to build and expand its content.
_The Turing Way_ guides are continually evolving, with multiple chapters co-developed by individuals from varied backgrounds - who are all passionate about sharing knowledge around data science and research.
To sustain and support this vibrant community, _The Turing Way_ book should remain consistent and accessible as it evolves. 
The {ref}`ch-style-guide` chapter already provides guidelines for maintaining a consistent style across the book. However, not all chapters follow these suggestions since they're often written asynchronously by different authors. 

> This pointed towards a need to encourage and support our contributors to maintain consistency throughout the chapters in _The Turing Way_ guides. 

Reviewer 3: ```suggestion
Reviewing consistency across _The Turing Way_ alone is reasonably challenging. This points towards a need to encourage and support our contributors to maintain consistency throughout the chapters in _The Turing Way_ guides. 
```

> Achieving this goal alone is reasonably challenging. 

Reviewer 3: ```suggestion
```
